Feature/server-info

megoth opened the pull request:
This PR solves #825 .

- Adds options to init script, that allows those setting up servers to give some human readable info to their visitors
- Makes use of that data to populate a section for the default server index page

In addition:

- It includes a bug fix (didn't copy all of the content in `default-templates/server` to the `data/[server dir]`)
- Some refactoring to reuse existing functionality
- Suggesting where to put util-methods (common methods that are reusable across the system)

megoth commented:
I've made this PR for `develop`, please let me know if that should be changed.

I've added three reviewers, might not need feedback from all.

RubenVerborgh commented on lib/server-config.js:
```diff
@@ -34,28 +37,29 @@ function ensureDirCopyExists (fromDir, toDir) {
+  // let defaultIndexPage = path.join(templates.server, 'index.html')
+  const existingIndexPage = path.join(serverRootDir, 'index.html')
+  // let defaultIndexPageAcl = path.join(templates.server, 'index.html.acl')
+  // let existingIndexPageAcl = path.join(serverRootDir, 'index.html.acl')
```
Remove commented-out code?

RubenVerborgh commented on lib/server-config.js:
```diff
@@ -34,28 +37,29 @@ function ensureDirCopyExists (fromDir, toDir) {
+      serverLogo: server ? server.logo : '',
+      serverVersion: parent._version
+    })
+    // fs.copySync(defaultIndexPageAcl, existingIndexPageAcl)
```
Remove commented-out code?

RubenVerborgh commented on lib/server-config.js:
```diff
@@ -34,28 +37,29 @@ function ensureDirCopyExists (fromDir, toDir) {
+  // let existingIndexPageAcl = path.join(serverRootDir, 'index.html.acl')
 
   if (!fs.existsSync(existingIndexPage)) {
+    console.log(argv)
```
Remove.

RubenVerborgh commented on lib/server-config.js:
```diff
@@ -34,28 +37,29 @@ function ensureDirCopyExists (fromDir, toDir) {
+      serverName: server ? server.name : host.hostname,
+      serverDescription: server ? server.description : '',
+      serverLogo: server ? server.logo : '',
+      serverVersion: parent._version
```
Not sure where `parent` comes from, but just wanted to remark that you can `require('../package.json')` in case you don't.

megoth commented:
I've cleaned up some comments and console.log and changed where version is fetched from (using require('package.json'), as per suggested by @RubenVerborgh ^_^

TallTed commented on bin/lib/options.js:
```diff
@@ -317,6 +318,22 @@ module.exports = [
+  },
+  {
+    name: 'server-logo',
+    help: 'A logo that represents you, your brand or your server (not required)',
```
```suggestion
    help: 'A logo that represents you, your brand, or your server (not required)',
```

megoth commented:
I've updated this so it can be merged into develop.

@kjetilk should we merge it now? I think it may be conflicting with https://github.com/solid/node-solid-server/pull/879 at some point. But I would really like to get this through, as there are things in it that I want to reuse for https://github.com/solid/node-solid-server/issues/799.

megoth commented:
@RubenVerborgh do you think the PR is good for merge?

RubenVerborgh commented:
No objections.

TallTed commented on default-templates/server/index.html:
```diff
@@ -27,6 +27,11 @@ <h1>Welcome to the Solid Prototype</h1>
 
+  <p class="lead hidden" id="loggedIn">
+    You are logged in as
+    <a href="#" id="profileLink"></a>.
```
This does not appear to show the user their logged in WebID, nor any label associated with that WebID.  I submit that both of these must be available -- not least because I may have the same label on multiple WebIDs, but only want to associate one WebID with some activity or data.
